cubzh-file-format-3zh.txt

```diff
@@ -0,0 +1,216 @@
+Cubzh .3zh File Format [10/20/2022]
+
+File
+{
+    Header
+
+    // PNG Preview
+    Chunk 'PREVIEW' : optional
+
+    Chunk 'NB_SHAPES' : N, optional
+
+    // Palette
+    Chunk 'PALETTE' : optional
+
+    // Objects (x N)
+    Chunk 'SHAPE' / Chunk 'OBJECT'
+
+    ...
+
+    Chunk 'SHAPE' / Chunk 'OBJECT'
+}
+-------------------------------------------------------------------------------
+
+
+1. Header
+-------------------------------------------------------------------------------
+# Bytes  | Type       | Value
+-------------------------------------------------------------------------------
+6        | byte       | magic bytes 'CUBZH!' : 'C' 'U' 'B' 'Z' 'H' '!', 'C' is first
+4        | int        | version number : 6
+1        | byte       | p3s compression method : 0 (none), 1 (zip), 2 (count)
+4        | uint       | total size
```

Comment:
Maybe we should precise that this is the total size of data(compressed or not) coming after this number. (but the way, it's a `uint32_t`)